File: Creational_Patterns/Singleton/Java/Lazy.java
class Lazy {
  private static Lazy instance = null;

  private Lazy() {}

  public static Lazy getInstance(){
    if(instance == null){
      instance = new  Lazy();
    }
    return instance;
  }
}

class Main{
  public static void main(String[] args) throws Exception{
    Lazy ob1 = Lazy.getInstance();
    Lazy ob2 = Lazy.getInstance();
    // Eager(); get error
    System.out.println(ob1 == ob2);
  }
}

File: Creational_Patterns/Singleton/python/Lazy.py

class LazySingleton:
  _instance = None

  def __init__(self):
    if LazySingleton._instance is not None:
      raise Exception("Alredy Existed Instance Don't create a new one")
    LazySingleton._instance = self
  
  @classmethod
  def get_instance(cls):
    if cls._instance is None:
      cls._instance = LazySingleton()
    return cls._instance

ob1 = LazySingleton.get_instance()
ob2 = LazySingleton.get_instance()

print(ob1 is ob2)